components/apimgt/org.wso2.carbon.apimgt.rest.api.store.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/store/v1/impl/ApisApiServiceImpl.java

```diff
@@ -1009,6 +1015,37 @@ public Response apisApiIdSubscriptionPoliciesGet(String apiId, String organizati
         return null;
     }
 
+    @Override
+    public Response getIdpsOfAPI(String apiId, MessageContext messageContext) {
+        try {
+            APIConsumer apiConsumer = RestApiCommonUtil.getLoggedInUserConsumer();
+            String organization = APIUtils.getOrganization(messageContext);
+            ApiTypeWrapper apiTypeWrapper = apiConsumer.getAPIorAPIProductByUUID(apiId, organization);
+            List<String> configuredIDPs = new ArrayList<>();
+            if (!apiTypeWrapper.isAPIProduct()) {
+                API api = apiTypeWrapper.getApi();
+                configuredIDPs = api.getKeyManagers();
+            }
+            APIAdmin apiAdmin = new APIAdminImpl();
+            List<KeyManagerConfigurationDTO> idpConfigurations = new ArrayList<>();
+            if (!configuredIDPs.isEmpty() && configuredIDPs.contains(APIConstants.KeyManager
+                    .API_LEVEL_ALL_KEY_MANAGERS)) {
+                idpConfigurations = apiAdmin.getKeyManagerConfigurationsByTenant(organization);
+            } else {
+                for (String idp : configuredIDPs) {
+                    idpConfigurations.add(apiAdmin.getKeyManagerConfigurationByName(organization, idp));
+                }
+            }
+            return Response.ok(KeyManagerMappingUtil.toKeyManagerListDto(idpConfigurations)).build();
+        } catch (APIMgtResourceNotFoundException e) {
+            RestApiUtil.handleResourceNotFoundError(RestApiConstants.RESOURCE_API, apiId, e, log);
+        } catch (APIManagementException e) {
```

Comment:
Why can't we through the APIManagementException to the Global Handler as done in [1]?

[1] https://github.com/wso2/carbon-apimgt/blob/master/components/apimgt/org.wso2.carbon.apimgt.rest.api.store.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/store/v1/impl/ApisApiServiceImpl.java#L1069

Reply:
Fixed in 32aae8fa15ab3a07e2e6896a1f702cfd91df1f05